File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Commons/ContainerProperty.cs
using System;

namespace VecompSoftware.DocSuiteWeb.Entity.Commons
{
    public class ContainerProperty : DSWBaseEntity
    {
        #region [ Constructor ]

        public ContainerProperty() : this(Guid.NewGuid()) { }
        public ContainerProperty(Guid uniqueId)
            : base(uniqueId)
        {
        }
        #endregion

        #region[ Properties ]
        public string Name { get; set; }
        public ContainerPropertyType ContainerType { get; set; }
        public int? ValueInt { get; set; }
        public DateTime? ValueDate { get; set; }
        public double? ValueDouble { get; set; }
        public bool? ValueBoolean { get; set; }
        public Guid? ValueGuid { get; set; }
        public string ValueString { get; set; }
        #endregion

        #region [ Navigation Properties ]
        public virtual Container Container { get; set; }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Common/ContainerProperty.vb

<Serializable()>
Public Class ContainerProperty
    Inherits DomainObject(Of Guid)
    Implements IAuditable

#Region " Constructors "
    Public Sub New()
        Id = Guid.NewGuid()
    End Sub

#End Region

#Region " Properties "

    Public Overridable Property Name() As String

    Public Overridable Property ContainerType() As ContainerPropertyType

    Public Overridable Property ValueInt() As Long?

    Public Overridable Property ValueDate() As DateTime?

    Public Overridable Property ValueDouble() As Double?

    Public Overridable Property ValueString() As String

    Public Overridable Property ValueGuid() As Guid?

    Public Overridable Property ValueBoolean() As Boolean?

    Public Overridable Property RegistrationUser() As String Implements IAuditable.RegistrationUser

    Public Overridable Property RegistrationDate() As DateTimeOffset Implements IAuditable.RegistrationDate

    Public Overridable Property LastChangedUser As String Implements IAuditable.LastChangedUser

    Public Overridable Property LastChangedDate As DateTimeOffset? Implements IAuditable.LastChangedDate

    Public Overridable Property Container() As Container

#End Region


End Class
